src/log-store/src/kafka/log_store.rs

```diff
@@ -199,6 +205,13 @@ impl LogStore for KafkaLogStore {
             ))
             .await?;
 
+        // Updates the high watermark offset of the last record in the topic.
+        for (region_id, offset) in &region_grouped_max_offset {
+            // Safety: `region_id` is always valid.
+            let provider = region_to_provider.get(region_id).unwrap();
+            self.high_watermark.insert(provider.topic.clone(), *offset);
+        }
```

Comment:
The watermark is only updated on writes. We might need to consider adding a periodic task to fetch the latest offset to keep the watermark up to date even when there are no writes.